### PR TITLE
auroradns: Add support for Health Checks

### PR DESCRIPTION
**Pull request by wido**
AuroraDNS supports Health Checks and based on the state of these
checks records will be served or not.

This way a Round-Robin DNS balancing can be achieved pointing to
only healthy servers/services.

### CODE REVIEW

File: libcloud/dns/drivers/auroradns.py [hunk 1]
```diff
@@ -140,6 +272,14 @@ def list_zones(self):
         return zones
 
     def list_records(self, zone):
+        """
```
Comment (Kami) [inline libcloud/dns/drivers/auroradns.py]: There is no need to add docstring for methods which are the same as the ones already defined in the base class - those docstrings are automatically inherited from the parent class methods.

File: docs/examples/dns/auroradns/health_checks.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from libcloud.dns.providers import get_driver
+from libcloud.dns.drivers.auroradns import AuroraDNSHealthCheckType
+
+cls = get_driver(Provider.AURORADNS)
+
+driver = cls('myapikey', 'mysecret')
+
+zone = driver.get_zone('auroradns.eu')
+
+health_check = driver.create_healthcheck(zone=zone,
```
Comment (Kami) [inline docs/examples/dns/auroradns/health_checks.py]: I assume this should be `ex_create_health_check`? :)

File: docs/examples/dns/auroradns/enable_disable_record.py [hunk 3]
```diff
@@ -1,6 +1,5 @@
-from libcloud.dns.types import Provider
+from libcloud.dns.types import Provider, RecordType
```
Comment (Kami) [inline docs/examples/dns/auroradns/enable_disable_record.py]: We tend to use one import per line - this makes searching and grepping a bit easier.

**Comment by wido** [comment]
@Kami Thank you for the quick reviews! I just force pushed a new version

File: libcloud/dns/drivers/auroradns.py [hunk 4]
```diff
@@ -222,6 +358,205 @@ def update_record(self, record, name, type, data, extra):
...
+        :type  threshold: ``int``
+
+        :param ipaddress: (optional) The IP Address of the target to monitor.
+                          You can pass a empty string if this is not required.
+        :type  ipaddress: ``str``
+
+        :param enabled: (optional) If this healthcheck is enabled to run
+        :type  enabled: ``bool``
+
+        :param extra: (optional) Extra attributes (driver specific).
```
Comment (jimbobhickville) [inline libcloud/dns/drivers/auroradns.py:437]: `extra` doesn't seem to be used.  Is it needed?  Are there other 'health check' drivers that use it and this is just allowing it to be passed in for consistency?

Comment (wido) [inline libcloud/dns/drivers/auroradns.py:437]: That is correct. It is indeed just there for consistency with the other APIs

A more generic approach for health checks might be nice, I think that Route 53 and Google DNS also support this.

**Comment by wido** [comment]
I updated the commit with tests for AuroraDNS. They were merged in #679 

Tests are passing on my system, waiting for Travis CI :)

**Comment by tonybaloney** [comment]
Thanks @wido

File: libcloud/dns/drivers/auroradns.py [hunk 5]
```diff
@@ -239,18 +572,34 @@ def __res_to_record(self, zone, record):
...
+                      type=record['type'],
+                      data=record['content'], zone=zone,
+                      driver=self, ttl=record['ttl'],
+                      extra=extra)
 
     def __res_to_zone(self, zone):
-        return Zone(id=zone['id'], domain=zone['name'], type=DEFAULT_ZONE_TYPE,
+        return Zone(id=zone['id'], domain=zone['name'],
+                    type=DEFAULT_ZONE_TYPE,
                     ttl=DEFAULT_ZONE_TTL, driver=self,
```
Comment (tonybaloney) [inline libcloud/dns/drivers/auroradns.py]: here too

File: libcloud/dns/drivers/auroradns.py [hunk 6]
```diff
@@ -239,18 +572,34 @@ def __res_to_record(self, zone, record):
...
+                                    type=healthcheck['type'],
+                                    hostname=healthcheck['hostname'],
+                                    ipaddress=healthcheck['ipaddress'],
+                                    health=healthcheck['health'],
+                                    threshold=healthcheck['threshold'],
+                                    path=healthcheck['path'],
+                                    interval=healthcheck['interval'],
+                                    port=healthcheck['port'],
+                                    enabled=healthcheck['enabled'],
+                                    zone=zone, driver=self)
```
Comment (tonybaloney) [inline libcloud/dns/drivers/auroradns.py]: I don't know how this method is used. but you should use self.connection.driver. If there are multiple drivers instantiated weird things would happen with this reference!

File: libcloud/dns/drivers/auroradns.py [hunk 7]
```diff
@@ -239,18 +572,34 @@ def __res_to_record(self, zone, record):
         extra['ttl'] = record['ttl']
         extra['prio'] = record['prio']
 
-        return Record(id=record['id'], name=name, type=record['type'],
-                      data=record['content'], zone=zone, driver=self,
-                      ttl=record['ttl'], extra=extra)
+        return Record(id=record['id'], name=name,
+                      type=record['type'],
+                      data=record['content'], zone=zone,
+                      driver=self, ttl=record['ttl'],
```
Comment (tonybaloney) [inline libcloud/dns/drivers/auroradns.py]: and here

**Comment by wido** [comment]
@tonybaloney Thanks for the feedback! I wasn't aware of that. Fixed that in a new commit.

**Comment by tonybaloney** [comment]
thanks. merging

**Pull request closed without merging**